src/Icicle/Sea/Eval.hs

```diff
@@ -0,0 +1,143 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PatternGuards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Icicle.Sea.Eval (
+    SeaError (..)
+  , seaEval
+  ) where
+
+import           Control.Monad.IO.Class (liftIO)
+import           Control.Monad.Trans.Either (EitherT(..), hoistEither)
+
+import           Data.Text (Text)
+import qualified Data.Text as T
+import           Data.Word (Word64)
+import qualified Data.Map as Map
+
+import           Foreign.C (CChar)
+import           Foreign.Marshal (allocaBytes, withArray, withArrayLen)
+import           Foreign.Ptr (Ptr, nullPtr)
+import           Foreign.Storable (Storable(..))
+
+import           Icicle.Avalanche.Prim.Flat (Prim)
+import           Icicle.Avalanche.Program (Program)
+import           Icicle.Common.Annot (Annot)
+import           Icicle.Common.Base
+import           Icicle.Common.Type (ValType(..))
+import           Icicle.Data
+import           Icicle.Data.DateTime (daysOfDate)
+import           Icicle.Internal.Pretty (Doc, Pretty, displayS, renderPretty)
+import           Icicle.Internal.Pretty (pretty, text)
+import           Icicle.Sea.FromAvalanche (seaOfProgram, outputsOfProgram)
+
+import           Jetski
+
+import           P hiding (count)
+
+import           System.IO (IO)
+
+------------------------------------------------------------------------
+
+data SeaError
+   = SeaJetskiError          JetskiError
+   | SeaValueConversionError Value
+   | SeaTypeConversionError  ValType
+   | SeaNoOutputs
+   deriving (Eq, Show)
+
+instance Pretty SeaError where
+   pretty (SeaValueConversionError v) = text "Cannot convert: " <> pretty v
+   pretty (SeaTypeConversionError  t) = text "Cannot convert: " <> pretty t
+   pretty (SeaNoOutputs)              = text "No outputs"
+
+   pretty (SeaJetskiError je) =
+     case je of
+       CompilerError _ _ stderr -> pretty stderr
+       _                        -> pretty (show je)
+
+------------------------------------------------------------------------
+
+seaEval :: (Show a, Show n, Pretty n, Ord n)
+         => Program (Annot a) n Prim
+         -> DateTime
+         -> [AsAt Value]
+         -> EitherT SeaError IO [(OutputName, Value)]
+
+seaEval program date values = do
```

Comment:
Yeah definitely, I think the whole interface to the repl needs a bit of an overhaul to make it friendlier to consume from ice